contrib/babelfishpg_tsql/sql/sys_procedures.sql

```diff
@@ -124,34 +134,47 @@ BEGIN
     RAISE EXCEPTION 'invalid option: %', "@option_scope";
   END IF;
 
-  SELECT COUNT(*) INTO cnt FROM pg_catalog.pg_settings WHERE name collate "C" like normalized_name and name collate "C" like '%escape_hatch%';
-  IF cnt = 0 THEN
+  SELECT COUNT(*) INTO cnt FROM sys.babelfish_configurations_view where name collate "C" like normalized_name;
+  IF cnt = 0 THEN 
     RAISE EXCEPTION 'unknown configuration: %', normalized_name;
+  ELSIF cnt > 0 AND (lower("@option_value") != 'ignore' AND lower("@option_value") != 'strict' 
+                AND lower("@option_value") != 'default' AND lower("@option_value") != 'on' AND lower("@option_value") != 'off') THEN
+    RAISE EXCEPTION 'unvalid option: %', lower("@option_value");
+  ELSIF cnt = (SELECT COUNT(*) FROM sys.babelfish_configurations_view) AND (lower("@option_value") = 'on' OR lower("@option_value") = 'off') THEN 
+    RAISE EXCEPTION 'unvalid option : %, when trying to use wildcard patterns.',lower("@option_value");
```

Comment:
This doesn't look correct because of the following cases:
- cnt = 1 and option_value = json
- cnt < SELECT COUNT(*) FROM sys.babelfish_configurations_view and option_value = on

Why don't we use
```
if cnt > 1 then option_value should be one of ignore/strict/default
```
?